bool boolean
int32 num
uint32 unsigned_num
float32 float_num
float64 double_num
string str